src/components/PostsList/PostsList.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../types';
+
+type Props = {
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
+};
+
+export const PostsList: React.FC<Props> = ({
+                                             userSelectedId,
+                                             selectPost,
+                                             post,
+                                           }) => {
```

Comment:
Indentation could be improved here. 

```suggestion
                                           }) => {
                                           export const PostsList: React.FC<Props> = ({
  userSelectedId,
  selectPost,
  post,
}) => {
```